nav2_rviz_plugins/src/nav2_panel.cpp

```diff
@@ -566,15 +994,82 @@ Nav2Panel::onCancelButtonPressed()
     }
   }
 
-
   timer_.stop();
 }
 
 void
 Nav2Panel::onAccumulatedWp()
 {
+  if (acummulated_poses_.size() == 0) {
+    state_machine_.postEvent(new ROSActionQEvent(QActionState::INACTIVE));
+    waypoint_status_indicator_->setText(
+      "<b> Note: </b> Uh oh! Someone forgot to select the waypoints");
+    return;
+  }
+
+  // Sanity check for the loop value
+  if (!isLoopValueValid(loop_no_)) {
+    state_machine_.postEvent(new ROSActionQEvent(QActionState::INACTIVE));
+    return;
+  }
+
+  // Remove any warning status at this point
+  waypoint_status_indicator_->clear();
+
+  // Disable navigation modes
+  navigation_mode_button_->setEnabled(false);
+  pause_resume_button_->setEnabled(false);
+
   std::cout << "Start waypoint" << std::endl;
+
+  // storing and removing the initial pose based on checkbox state
+  if (store_poses_.size() == 0) {
+    // Setting the final loop value on the text box for sanity
+    nr_of_loops_->setText(QString::fromStdString(loop_no_));
+
+    // Variable to store initial pose
+    geometry_msgs::msg::TransformStamped init_transform;
+
+    // Looking up transform to get initial pose
+    if (store_initial_pose_) {
+      try {
+        init_transform = tf2_buffer->lookupTransform(
+          "map", "base_footprint",
```

Comment:
I don't suppose there are other things in this plugin we can base these off of instead of hard coding them? Maybe somewhere something coming in has a map and/or base_footprint frame that can be stored for use here?

Reply:
We can definitely get the `map` frame.. But I'm not sure if there is a way to get the base frame..

Reply:
OK - lets at least do map, I see other functions for navigation before this PR gets map frame to send navigation requests. 

Maybe expose the base_frame as a parameter so that it can be overrided if someone likes, but yeah I doubt anyone will functionally do that. More about making it possible.

Reply:
Sure, I'll do that.. 

-----------

Maybe somewhere in the future, we might want to have a single RViz for multi-robot system, then we might need a huge overhaul of the complete plugin with the namespaces.. 

We already have a [plugin](https://github.com/neobotix/neo_fleet_rviz2_plugin/tree/cleanup) for selecting a robot and sending a goal for that particular robot.. Of course, all done in a single RViz.. but it certainly lacks the features that are present right now in the nav2_plugin..

Reply:
> we might want to have a single RViz for multi-robot system

That's been suggested before and I've generally been opposed to it being a requirement / task for maintainers. If someone made a new rviz plugin for multi-robot, I'd be happy to review/merge it, but its not the kind of thing that the maintainers are going to work on and its a pipe-dream feature request in the issue tracker unless the person proposing it is going to do it themselves. I try to keep the issue tracker to actionable features (and obviously all bugs) which will realistically be accomplished eventually